src/Sulu/Bundle/AdminBundle/Resources/js/components/DatePicker/DatePicker.js

```diff
@@ -0,0 +1,202 @@
+// @flow
+import React from 'react';
+import ReactDOM from 'react-dom';
+import type {ElementRef} from 'react';
+import ReactDatetime from 'react-datetime';
+import {observer} from 'mobx-react';
+import {action, observable} from 'mobx';
+import 'react-datetime/css/react-datetime.css';
+import type Moment from 'moment';
+import moment from 'moment';
+import Input from '../Input';
+import Popover from '../Popover';
+import './datePicker.scss';
+
+type Props = {
+    value: ?Date,
+    /** Placeholder can be overwritten */
+    placeholder?: string,
+    /** Listen for changes of this component */
+    onChange: (value: ?Date) => void,
+    /** Configure the datepicker to your needs, for more information have a look in the README.md */
+    options: Object,
+    valid: boolean,
+};
+
+@observer
+export default class DatePicker extends React.Component<Props> {
+    static defaultProps = {
+        options: {},
+        valid: true,
+    };
+
+    inputChanged: boolean = false;
+
+    @observable open: boolean = false;
+    @observable showError: boolean = false;
+    @observable value: ?string | ?Date | ?Moment = null;
+    @observable inputRef: ?ElementRef<*>;
+
+    @action setOpen(open: boolean) {
+        this.open = open;
+    }
+
+    @action setValue(value: ?string | ?Date | ?Moment) {
+        this.value = value;
+    }
+
+    @action setShowError(showError: boolean) {
+        this.showError = showError;
+    }
+
+    @action setInputRef = (ref: ?ElementRef<*>) => {
+        this.inputRef = ref;
+    };
+
+    componentWillMount() {
+        this.setValue(this.props.value);
+    }
+
+    componentWillReceiveProps(nextProps: Props) {
+        if (this.value && !nextProps.value) {
+            return;
+        }
+
+        this.setValue(nextProps.value);
+    }
+
+    handleChange = (date: ?Date) => {
+        this.inputChanged = false;
+        this.props.onChange(date);
+
+        this.setShowError(!!this.value && !date);
+    };
+
+    handleDatepickerChange = (date: string | Moment) => {
+        if (!date) {
+            this.setValue(undefined);
+            this.handleChange(undefined);
+
+            return;
+        }
+
+        if (typeof date === 'string') {
+            this.setValue(date);
+
+            return;
+        }
+
+        if (!date.isValid()) {
+            this.handleChange(undefined);
+
+            return;
+        }
+
+        this.handleChange(date.toDate());
+    };
+
+    handleInputBlur = () => {
+        if (this.inputChanged && typeof this.value === 'string') {
+            this.handleChange(undefined);
+        }
+    };
+
+    handleOpenOverlay = () => {
+        this.setOpen(true);
+    };
+
+    handleCloseOverlay = () => {
+        this.setOpen(false);
+    };
+
+    getInputChange = (props: Object) => {
+        return (value: string | Moment) => {
+            const e = {
```

Comment:
Seems like @chirimoya forgot to add a comment here: We think it would be better to add the real DOM event of the `Input` here. That means that the `onChange` of the `Input` would have to add it as a second parameter. But then we would not have to fake a DOM event here.

Reply:
e.g. https://github.com/react-toolbox/react-toolbox/blob/dev/components/input/Input.js#L111